#5071 Implement General Feedback Modal in TDP Frontend

lhawkman27 opened the pull request:
## Summary of Changes
- Added in "Give Feedback" button in the bottom right corner of the viewport of the application to open Feedback Modal with a Feedback Form within.
- Added functionality for users to provided feedback by selecting a rating and entering a "Tell us more" message.
- Selecting a rating is required to submit feedback, otherwise an error message will display as well as other error state changes , until user selects a rating. 
- Added a anonymous checkbox in the Feedback form so users can anonymously submit feedback when checked.
- Once user submits feedback, the Feedback Modal's text changes to saying "Thank you for your Feedback" as well as providing users with an email if they want to talk with support. 

Pull request closes #5071 

## How to Test
```
cd tdrs-frontend && docker-compose up --build
cd tdrs-backend && docker-compose up --build
```

1. Open http://localhost:3000/ and sign in
2. Click the "Give Feedback" button in the bottom right corner.
3. Should see Send Anonymously checkbox.
4. When Modal opens -> Click "Send Feedback" button. (Users can click or tab to selection and hit ENTER)
5. View Error state changes. 
6. Next select a User Experience rating. (Users can click or tab to selection and hit ENTER)
7. View state change back to normal.
8. Now click "Send Feedback" button.
9. View the change of the Modal to the "Thank you for you feedback" message version. 
10. To close Modal click "Close" button or the "X" button at the top of modal. (click or tab to and hit ENTER)
11.  Open Modal again (Modal should be reset)
12. Now you can selecting a rating and enter a feedback message
13. Check Send anonymously (click or tab to and hit ENTER)
14. Click "Send Feedback" again (should show "Thank you message again")
15. Sign out if you haven't already and you should not see the Send Anonymously checkbox anymore. 
16. Repeat step 12 and while focused on selected rating or text area hit CMD+Enter and form should properly submit and show Thank you modal

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [ ] “Give Feedback” button appears on all application pages, fixed to the lower-right viewport
+ [ ] “Give Feedback” button appears on all application pages, fixed to the lower-right viewport  
+ [ ] Modal matches UX mockups in structure, spacing, and interaction behavior  
+ [ ] Form includes:
  - Required validation for free-text input (if defined by UX)
  - Optional anonymity toggle
  - Valid Likert input submission  
+ [ ] Modal closes on successful submission and shows confirmation feedback  
+ [ ] Graceful error handling for backend failures, timeouts, or network issues  
+ [ ] Jest unit tests implemented and passing  
+ [ ] Testing checklist executed with no blocking issues  
+ [ ] README updated (if new dependencies, env vars, or setup steps are introduced)  
+ [ ] [Release digest](https://app.gitbook.com/o/Yd4Wv0Fi89kSKpmakWrn/s/JSIbkxujKUxhKv2cXFEg/) entry added with screenshots, behavior summary, and testing steps
+ [ ] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?


### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [ ] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [ ] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

Commit:
#5701 finished up with custom component creations, and adding sticky button to bottow of apps viewport. Added feelings icon for feedback form

Commit:
#5701 updating code to pass information between components properly and added in sticky button and it's functionality. Adding validation error color and text as well

Commit:
#5071 fixing up UI elements and testing the Ui functionality with simple validation

Commit:
#5071 refactored some of my code after some more testing, wrote some unit tests as well

Commit:
#5071 finished up jest and cypress test, and also cleaned up the code

Commit:
Merge branch 'develop' into 5071-implement-general-feedback-modal-in-tdp-frontend

Commit:
#5071 removed console log

Commit:
#5071 fixed cypress test

lhawkman27 commented:
![Screenshot 2025-06-12 at 11 37 14 AM](https://github.com/user-attachments/assets/41b62e23-9728-4bc0-98b0-c4b135808d37)
![Screenshot 2025-06-12 at 11 37 38 AM](https://github.com/user-attachments/assets/7f18365f-c573-4ed0-8cff-6677e4f4bf98)
![Screenshot 2025-06-12 at 11 38 23 AM](https://github.com/user-attachments/assets/f630ca55-ab2c-49e2-b5ba-0ee55bec2d79)
![Screenshot 2025-06-12 at 11 39 10 AM](https://github.com/user-attachments/assets/8b728914-3f59-4959-97ee-fc7072385c86)
![Screenshot 2025-06-12 at 12 11 47 PM](https://github.com/user-attachments/assets/dedf7ce4-c72c-4e14-b179-ecced57f5569)

mattcoleanderson commented:
@lhawkman27 hey man you added us all as **assignees**. What you meant to do was add us as **reviewers**. You should be the only assignee since you did the work.

I went ahead and corrected it for you.

mattcoleanderson commented on tdrs-frontend/jsconfig.json:
This is a VSCode configuration file. We probably want to add this to .gitignore and untrack it.

lhawkman27 replied:
Nice catch I added it to the .gitignore.

mattcoleanderson replied:
Looks like you haven't pushed any changes for .gitignore

lhawkman27 replied:
that's my bad I kept skipping over that file when pushing (out of habit), the changes to the file are pushed up now

lhawkman27 replied:
And removed the jsconfig.json file from being tracked anymore, should be gone now

Commit:
#5071 prettier fix for test

mattcoleanderson commented on tdrs-frontend/src/App.js:
This is the root component for the frontend. I suggest you move as much of the feedback modal logic into a sub component as possible to keep the root component clean and modular.

Most of this logic should be able to live inside of your FeedbackModal and the html components could be injected via a `<FeedbackModal />` tag.

lhawkman27 replied:
I actually debated this when I was writing the code, but I wan't sure about it. But I created the Feedback component which handles the logic for the modal displaying or not. So the App.js is clean and only calls the Feedback component now.

mattcoleanderson reviewed (dismissed):
Solid work. I have a couple suggestions for code organization.

elipe17 reviewed (changes requested):
I am able to escape the modal with tabs and do things in the background. The Modal component in components dir has a way to ensure the tabs don't escape.

<img width="1008" alt="Screenshot 2025-06-13 at 10 47 03 AM" src="https://github.com/user-attachments/assets/394fa594-6242-48f0-8e18-5e8adcdac1f5" />

Commit:
#5071 refactored App.js and pulled modal logic into a helper component Feedback.jsx, updated tabbing logic for FeedbackModal and Form, added in validation check for malicious text input, added more unit tests for coverage (please work), and fixed cypress test (also please work)

lhawkman27 commented:
@elipe17 thanks for the tab test I overlooked that. I added in some refined tabbing logic to stop that behavior, and also I added in a sanitize function for the textarea input to check for <script>...</script> entered tags as well.

jtimpe commented on tdrs-frontend/src/components/FeedbackRadioSelectGroup/FeedbackRadioSelectGroup.jsx:
```diff
@@ -0,0 +1,159 @@
+}) => {
+  return (
+    <div
+      style={{
```
the inline style works, but i think the code would be cleaner if you moved the style to an external stylesheet (or sass file) and referenced the classes.

lhawkman27 replied:
100% agree, I moved the styling to a .scss file

jtimpe replied:
looks good!

jtimpe commented on tdrs-frontend/src/components/FeedbackRadioSelectGroup/FeedbackRadioSelectGroup.jsx:
```diff
@@ -0,0 +1,159 @@
+                border:
+                  selectedOption === option.value
+                    ? `1px solid ${option.value}`
+                    : '1px solid black',
```
another thing that slightly bothers my over-particular brain - the lines in between feedback rating options are double thickness compared to the outside lines, since you have a solid box around. This is easier if you move the style code to stylesheets too, but you could `border-right: 0px` and on the last element `border-right: 1px`. with stylesheets you can use psueudo-classes [like `:last-child`](https://developer.mozilla.org/en-US/docs/Web/CSS/:last-child)

the highlighted border might also look better if it was "inside" the button instead of outside, which you can [accomplish with `border-box`](https://stackoverflow.com/a/9601431), that way the size of the button doesn't change when hovered/selected.

these are tiny details, so feel free to ignore, just thought i'd throw out a few ideas 😄

lhawkman27 replied:
fixed the styling issue with the border, so I think it should look good now.

jtimpe replied:
i don't see `border-box` or `border-right: 0px` - i'll leave this open and ux can weigh in. i think this may also contribute to the bounce/flicker effect eric pointed out

jtimpe commented on tdrs-frontend/src/components/FeedbackForm/FeedbackForm.jsx:
```diff
@@ -0,0 +1,171 @@
+import FeedbackRadioSelectGroup from '../FeedbackRadioSelectGroup'
+import { feedbackPost } from '__mocks__/mockFeedbackAxiosApi'
+
+const FeedbackForm = ({ onFeedbackSubmit }) => {
```
For the components generally: i prefer if "used together" components are colocated in the codebase. a recent example is the FRA Reports component (while not perfectly organized, only shared components are separated into separate folders). My suggestion for you is to move the separate files into the same `Feedback` or `FeedbackModal` folder if they aren't generic components.

Long term, I'd like to get to `pages` and `components` being separate directories so that's the kind of separation i'm thinking about

lhawkman27 replied:
I went ahead and moved the Feedback component to under the Feedback Modal folder.

jtimpe commented on tdrs-frontend/src/components/FeedbackForm/FeedbackForm.jsx:
```diff
@@ -0,0 +1,171 @@
+  }
+
+  const submitForm = (data) => {
+    return axios.post('/api/userFeedback/', data)
```
maybe a note for the api implementer, we probably want to use `axiosInstance` here to carry over the authentication credentials. related: #3722

lhawkman27 replied:
added a NOTE to utilize the axiosInstance when real api is being called

jtimpe commented on tdrs-frontend/src/components/FeedbackRadioSelectGroup/FeedbackRadioSelectGroup.jsx:
```diff
@@ -0,0 +1,159 @@
+                  key={option.value}
+                  className="radio-icon"
+                >
+                  <input
```
is it possible to make the radio/feedback options tab targets? tabbing through now focuses only on modal elements, but the smiley options aren't selectable.

lhawkman27 replied:
Made the radio buttons tabbable (arrow nav) based off native behavior

- Radios now behave like native HTML radios — only the selected radio is tabbable.
- Keyboard tab navigation will naturally go to the selected radio first.
- Arrow keys will cycle between radios properly (browser built-in).

jtimpe replied:
how do you select an option? tabbing seems to hover over an option, but if i hit "enter" and tab to the text input, i lose the selection.

lhawkman27 replied:
Added in the ability to select the radio options by keyDown on ENTER

jtimpe commented on tdrs-frontend/cypress/e2e/feedback/user-feedback.feature:
```diff
@@ -0,0 +1,6 @@
+    When user visits the home page
+    Given user clicks on Give Feedback button on home page
+    Then feed back modal and form should display to user
+    Then user attempts to submit invalid feedback and then valid feedback
```
```suggestion
    When the user attempts to submit valid feedback
    Then the user receives an error
```
my preference code-wise is to split up the action and assertion steps just to make the implementations more digestible.

open to suggestions here as we're still defining our e2e test strategy - but i also think covering only the success case in the cypress tests is sufficient (as long as other cases are sufficiently unit tested, which i see they are)

lhawkman27 replied:
Split out the cypress test one for success submission and one for attempt invalid submit

jtimpe replied:
couple things here - more just as a point of discussion, not necessarily requesting changes.
1. even though your `When` step has an assertion inside the implementation, your `.feature` file doesn't have an assertion step. I'd recommend having a separate `When` and `Then` in your test. I think you can also remove the `Then feed back modal and form should display to user` step and implement it as a line within the `When user clicks on Give Feedback` step implementation
2. you have one test that covers two cases, the valid and invalid submissions. the cypress tests feel a little slow and heavy (they are) but i'd still recommend splitting the tests into individual `Scenario`s rather than steps in the same scenario. the setup will have to run again, but it's actually faster once we enable parallelization. but more importantly, it makes gives the feature file nice separation and makes it easy to read. if slow tests are a concern, i'd recommend removing the "invalid" scenario entirely and just going with the happy paths, any relying on the unit test coverage for the failure cases.

```gherkin
When user visits the home page
    And user clicks on Give Feedback # <-- this impl checks for form visibility before proceeding to next step
    And user attempts to submit valid feedback
Then user sees a thank you message

When user visits the home page
    And user clicks on Give Feedback button
    And user attempts to submit invalid feedback
Then user sees an error message
```

check out [this article on gherkin](https://cucumber.io/docs/bdd/better-gherkin) - if we go with a more declarative style and treat the `.feature` file as a "plain english" description of the behavior of the feature, the tests start to look/read almost like documentation

hopefully some of that makes sense. happy to chat about it on a call as well!

jtimpe replied:
we might also consider another test for logged in users instead of just anonymous. and/or ticking the "anonymous" checkbox, although that could fall under a unit test.

jtimpe replied:
here's another article on cucumber antipatterns - https://www.thinkcode.se/blog/2016/06/22/cucumber-antipatterns. not able to link a section, so look for headers:
* Hard to tell what you are testing
* No clear separation between `Given`/`When`/`Then`
* Multiple `Then` in the same scenario

jtimpe commented on tdrs-frontend/src/components/FeedbackForm/FeedbackForm.test.js:
```diff
@@ -0,0 +1,261 @@
+  }
+})
+
+describe('Feedback Form tests', () => {
```
great job with these!

lhawkman27 replied:
Thanks!!! I fought with the test coverage for a while. I think I need retire from writing tests, I'm getting to old for those!!!!

jtimpe commented on tdrs-frontend/src/components/FeedbackForm/FeedbackForm.jsx:
```diff
@@ -0,0 +1,171 @@
+    setIsAnonymous(false)
+  }
+
+  function sanitizeInput(input) {
```
others can disagree, but i personally don't think this sanitization step is necessary. django already provides us [XSS and SQLi protection by default](https://docs.djangoproject.com/en/5.2/topics/security/#cross-site-scripting-xss-protection) when inserting into any text-based field. the escaping is also not super user friendly as it replaces escaped characters while you type

![image](https://github.com/user-attachments/assets/5a9f178f-f37d-4a77-b54a-5939683a7f40)

lhawkman27 replied:
Removed the sanitize function and that escape behavior is resolved now

jtimpe reviewed (changes requested):
Really good job overall! I have mostly organization-related comments and a couple UX enhancements. I think the santization-while-typing is really the biggest.

Commit:
#5071 fixed code coverage ???

Commit:
Merge branch 'develop' into 5071-implement-general-feedback-modal-in-tdp-frontend

Commit:
#5071 moved FeedbackForm and FeedbackRadioSelectGroup under FeedbackModal folder, updated unit tests, refactored cypress test to have sucess test and invalid tests, made an external stylesheet for feedback form, made radio tabbable

Commit:
#5071 fixed cypress tests

Commit:
#5071 update to cypress apparently vs code didnt save my change to the actual test file

Commit:
#5071 hopefully last clean up and fix for cypress test

Commit:
#5071 cypress fix

Commit:
5071 cypress fix

Commit:
#5071 testing to see if success case works in general

Commit:
#5071 fixes for tests

Commit:
#5071 fixed visual cue for tabbing on radio selects

Commit:
Merge branch 'develop' into 5071-implement-general-feedback-modal-in-tdp-frontend

Commit:
Merge branch 'develop' into 5071-implement-general-feedback-modal-in-tdp-frontend

elipe17 commented on tdrs-frontend/src/components/FeedbackModal/Feedback.jsx:
```diff
@@ -0,0 +1,98 @@
+        onClick={handleOpenModal}
+      >
+        Give Feedback
+      </button>
```
Is it possible to update this so that it makes use of the `Button` component? If not, can we style this so that it matches the rest of the buttons in the app? E.g. rounded corners. Also can we move this a few more pixels towards the right border? It feels weird to me that it's border lands on directly on the home page card borders. See below:

<img width="693" alt="Screenshot 2025-06-17 at 1 15 30 PM" src="https://github.com/user-attachments/assets/b977dcac-2fff-454f-8cab-8fad099d7504" />

lhawkman27 replied:
Moved the button over to the right a little more. Updated to using the Button component and fixed styling as needed. 

Refactored inline styling and moved to an external file in general

elipe17 commented on tdrs-frontend/src/components/FeedbackModal/FeedbackRadioSelectGroup.jsx:
```diff
@@ -0,0 +1,134 @@
+  },
+]
+
+const FeedbackRadioSelectGroup = ({
```
Nit: can we lighten up the stroke weight on the containers for the icons?

lhawkman27 replied:
Made the borders a bit smaller

elipe17 commented on tdrs-frontend/src/components/FeedbackModal/Feedback.jsx:
```diff
@@ -0,0 +1,98 @@
+  const [isFeedbackSubmitted, setIsFeedbackSubmitted] = useState(false)
+
+  const handleOpenModal = () => {
+    console.log('Opening feedback modal')
```
```suggestion
```

lhawkman27 replied:
removed

elipe17 commented on tdrs-frontend/src/components/FeedbackModal/Feedback.jsx:
```diff
@@ -0,0 +1,98 @@
+
+  const handleOnFeedbackSubmit = () => {
+    setIsFeedbackSubmitted(true)
+    console.log('Thank you for your feedback!')
```
```suggestion
```

lhawkman27 replied:
removed

elipe17 commented on tdrs-frontend/src/components/FeedbackModal/FeedbackForm.jsx:
```diff
@@ -0,0 +1,166 @@
+        if (response.status === 200) {
+          onFeedbackSubmit()
+          resetStatesOnceSubmitted()
+          console.log('Feedback submitted successfully!')
```
```suggestion
```

lhawkman27 replied:
removed

elipe17 commented on tdrs-frontend/src/components/FeedbackModal/FeedbackForm.jsx:
```diff
@@ -0,0 +1,166 @@
+          resetStatesOnceSubmitted()
+          console.log('Feedback submitted successfully!')
+        } else {
+          console.log('Something went wrong. Please try again.')
```
Should this render an error modal instead, or at least console.error?

lhawkman27 replied:
I made this console error

elipe17 commented on tdrs-frontend/src/components/FeedbackModal/FeedbackForm.jsx:
```diff
@@ -0,0 +1,166 @@
+      }
+    } else {
+      setError(true)
+      console.log('Please select a rating before submitting.')
```
```suggestion
```

lhawkman27 replied:
removed

elipe17 commented on tdrs-frontend/src/components/FeedbackModal/FeedbackModal.jsx:
```diff
@@ -0,0 +1,186 @@
+        'Active element:',
+        activeEl.dataset.testid || activeEl.outerHTML
+      )
+      console.log('Tab pressed: currentIdx', currentIdx, 'nextIdx', nextIdx)
```
```suggestion
```

lhawkman27 replied:
removed

elipe17 commented on tdrs-frontend/src/components/FeedbackModal/FeedbackModal.jsx:
```diff
@@ -0,0 +1,186 @@
+              focusableElements.length
+            : (currentIdx + 1) % focusableElements.length
+
+      console.log(
```
```suggestion
```

lhawkman27 replied:
removed

Commit:
#5071 cleaned up console logs and inline styling as well

Commit:
#5071 needed to update test do to changing buttons to utilize Button component due to it overridding data-testids, so need to get by text now

Commit:
Merge branch 'develop' into 5071-implement-general-feedback-modal-in-tdp-frontend

Commit:
#5071 updated test to look for error console logs

Commit:
#5071 updated some more test for coverage

Commit:
#5071 updated test for coverage

Commit:
Merge branch 'develop' into 5071-implement-general-feedback-modal-in-tdp-frontend

elipe17 commented on tdrs-frontend/src/components/FeedbackModal/FeedbackModal.jsx:
```diff
@@ -0,0 +1,172 @@
+        e.preventDefault()
+        onTabPressed(e.shiftKey)
+      } else if (e.key === 'Escape') {
+        e.preventDefault()
```
Should this close the modal?

lhawkman27 replied:
Do we want that behavior?? I thought the close (x) was going to be the only way to close it. If we want that I can update the code for that.

lhawkman27 replied:
@reitermb what's your opinion on this behavior for the feedback modal?

lhawkman27 replied:
Added in the ability to close the Modal by hitting ESC. It makes sense to have that in anyways, so I just added it in.

elipe17 commented on tdrs-frontend/src/components/FeedbackModal/FeedbackModal.jsx:
```diff
@@ -0,0 +1,172 @@
+  const hasAutoFocused = useRef(false)
+
+  useEffect(() => {
+    if (isOpen && modalRef.current && !hasAutoFocused.current) {
```
There seems to be some contention going on here between the browsers auto focus and this logic. It is causing the modal to flicker or hold its selected outline when any child elements are clicked.

https://github.com/user-attachments/assets/4d8f1b39-ba1a-482d-9cfc-5623cda0734e

lhawkman27 replied:
Alright fixed the flickering focus issue. I had tabeIndex on the overlay and modal content and it was causing it to flicker focus between both of them every time inputs were clicked. And fixed some styling issues as well for it

elipe17 replied:
I just pulled and built this again and I am still seeing the flickering and cannot close the modal with the escape key. I cleared my browser cache and tried an incognito window with no success. Perhaps you can demo it in OH today or we can pair to see if it is me?

jtimpe reviewed (approved):
Left a couple of my previous comments open with a few more points of discussion. Everything functions great though. Good work!

Commit:
#5071 updated .gitignore file

Commit:
#5071 added in functionality for selecting radio inputs and checkboxes with enter key down event. still working on styling issue with refocusing on overlay when items inside it are selected

Commit:
#5071 added in ability to close modal on escape key press (makes sense to have)

Commit:
#5071 fixed flicker focus on modal by removing conflicting tabIndex off off overlay and model content. Refactored some code and added in some better styling

Commit:
#5071 updated cypress tests to be two separate test cases

Commit:
Merge branch 'develop' into 5071-implement-general-feedback-modal-in-tdp-frontend

Commit:
#5071 fix for cypress test

Commit:
Merge branch 'develop' into 5071-implement-general-feedback-modal-in-tdp-frontend

lhawkman27 commented:
@reitermb and @kristen-shaw4 what's your opinions on users being able to hit the enter key as another means for submitting the feedback form?? Or even hitting CMD+ENTER to submit the form?

elipe17 reviewed (changes requested):
Ok I pulled and rebuilt and this time the flicker is gone and the escape/enter stuff works. Like we discussed in OH today, we'd like to see the following:

- Anonymous checkbox not shown when user is not authenticated
- Clicking enter on any element (except the submit button) should not submit the form. Consider CMD+Enter to submit
- Move the color border on the radio icons to the interior to avoid that layout shift

Commit:
#5071 updated styling and added in check for logined in user or not to control send anonyumous checkbox render, checking how cmd+enter works

Commit:
#5071 latest updates per OH feedback, add CMD+ENter submission, fixed icon radio styling

lhawkman27 commented:
- Added in logic for anonymous checkbox not showing if user isn't signed in.
- updated radio icon selection groups styling to fix layout shifting and border issues
- added in functionality to submit form when focused and user hits CMD+Enter

Commit:
Merge branch 'develop' into 5071-implement-general-feedback-modal-in-tdp-frontend

elipe17 reviewed (approved):
Nice job sir! One last thing would be to deploy it and double check everything works in a deployed environment, otherwise LGTM!

Commit:
Remove tracked file that should be ignored

Commit:
#5071 needed to update imports since jsconfig.json file was removed

Commit:
Merge branch 'develop' into 5071-implement-general-feedback-modal-in-tdp-frontend

Commit:
#5071 UI review style fixes

Commit:
Merge branch 'develop' into 5071-implement-general-feedback-modal-in-tdp-frontend

Commit:
#5071 fixed text color change per UI review

Commit:
Merge branch 'develop' into 5071-implement-general-feedback-modal-in-tdp-frontend

Commit:
Merge branch 'develop' into 5071-implement-general-feedback-modal-in-tdp-frontend
